Sources/Testing/ProcedureKitTestCase.swift

```diff
@@ -76,4 +116,38 @@ open class ProcedureKitTestCase: XCTestCase {
             }
         }
     }
+
+    // MARK: Stress Tests
+
+    public func stress(atLevel level: StressLevel = .medium, withName name: String = #function, withTimeout timeoutOverride: TimeInterval? = nil, block: (Int, Int, DispatchGroup) -> Void) {
+        let stressTestName = "Stress Test: \(name)"
+        let timeout: TimeInterval = timeoutOverride ?? level.timeout
+
+        print("\(stressTestName), Parameters: \(level.batches) batches, size \(level.batchSize), timeout: \(timeout)")
+
+        let overallDispatchGroup = DispatchGroup()
+        weak var overallExpectation = expectation(description: stressTestName)
+
+        queue.delegate = nil
+
+        (0..<level.batches).forEach { batch in
+            autoreleasepool {
+                let batchStartTime = CFAbsoluteTimeGetCurrent()
+                (0..<level.batchSize).forEach { iteration in
+                    block(batch, iteration, overallDispatchGroup)
+                }
+
+                let batchFinishTime = CFAbsoluteTimeGetCurrent()
+                let batchDuration = batchFinishTime - batchStartTime
+                print("\(stressTestName), finished batch: \(batch), in \(batchDuration) seconds")
+            }
+        }
+
+        overallDispatchGroup.notify(queue: .main) {
```

Comment:
The way that most of the current StressTests work is to utilize a dispatch group for each batch, and wait for each batch to complete. (i.e. Within the autoreleasepool.)

This waits after all batches.

From a timing standpoint, I found it more useful to have this at the batch level because the XCTest framework already provides a time for the completion of the entire test case.

Additionally, waiting at the end of each batch helps ensure that the autoreleasepool cleans up after all the finished operations in each batch (which helps keep memory usage bounded and predictable - and, thus, makes it easy to run as many batches as you want, bounded only by how long you feel like running the StressTests).